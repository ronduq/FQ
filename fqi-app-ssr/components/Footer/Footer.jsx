import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon/Icon';
import s from './Footer.scss';


const Footer = ({
  theme,
  footerItems
}) => {
  return (
    <footer className={classNames('container', s.footer)}>
      <a href="https://www.teakorigin.com" target="_blank" aria-label="Teak Origin (opens in new window)">
        <Icon className={classNames(s.logo, s.themeBlack, {[s.themeWhite]: theme === 'white'} )} iconName="logo" />
      </a>
      <ul>
        {footerItems.map(({link, title}, idx) => 
          <li key={`footer-${idx}`}>
            <a href={link} target="_blank" className={classNames({[s.themeWhite]: theme === 'white'} )}>
              {title}
              <span className="screen-reader-text">(opens in new window)</span>
            </a>
          </li>
        )}
      </ul>
    </footer>
  );
};

Footer.propTypes = {
  footerItems: PropTypes.array,
};

Footer.defaultProps = {
  theme: 'black',
};

export default Footer;
