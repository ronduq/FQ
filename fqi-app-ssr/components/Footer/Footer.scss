@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.footer {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 16px 0 16px;
  max-width: 100%;
  z-index: 1;
  color: $colour-grey-dark;

  .logo {
    height: 29px;
    width: 300px;
    display: block;

    @include respond-to('tablet') {
      height: 19px;
      width: 196px;
    }
  }

  ul {
    margin-top: 24px;

    @include respond-to('tablet') {
      margin: 32px 0 0
    }

    li {
      margin-right: 32px;
      display: inline-block;
      @include font-size('medium');
      @include font-family('proxima-bold');

      @include respond-to('tablet') {
        display: block;
        text-align: center;
        margin: 0 0 16px 0;
      }

      &:last-child {
        margin: 0;
      }
    }
  }
}