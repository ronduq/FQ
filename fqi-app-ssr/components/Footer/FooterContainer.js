import { connect } from 'react-redux'
import Footer from './Footer';
import { getFooterItems } from '../../selectors/index';


const mapStateToProps = state => {
  return {
    footerItems: getFooterItems(state),
  }
}

export default connect(
  mapStateToProps,
  null
)(Footer);