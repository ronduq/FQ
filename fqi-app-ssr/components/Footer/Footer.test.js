/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Footer from './Footer'

describe('With Snapshot Testing', () => {
  const props = {
    footerItems: [{link: 'www.aaa.com', title: 'link 1'}, {link: 'www.bbb.com', title: 'link 2'}]
  }
  it('Footer snapshot"', () => {
    const component = renderer.create(<Footer { ...props } />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

})