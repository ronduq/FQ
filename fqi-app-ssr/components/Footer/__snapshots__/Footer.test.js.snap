// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Footer snapshot" 1`] = `
<footer
  className="container footer"
>
  <a
    aria-label="Teak Origin (opens in new window)"
    href="https://www.teakorigin.com"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      className="logo themeBlack"
      focusable="false"
    >
      <use
        xlinkHref="#logo"
      />
    </svg>
  </a>
  <ul>
    <li>
      <a
        className=""
        href="www.aaa.com"
        target="_blank"
      >
        link 1
        <span
          className="screen-reader-text"
        >
          (opens in new window)
        </span>
      </a>
    </li>
    <li>
      <a
        className=""
        href="www.bbb.com"
        target="_blank"
      >
        link 2
        <span
          className="screen-reader-text"
        >
          (opens in new window)
        </span>
      </a>
    </li>
  </ul>
</footer>
`;
